symtabAPI/src/Object-elf.C

```diff
@@ -4184,3 +4256,211 @@ bool ObjectELF::getRegValueAtFrame(Dyninst::Address pc, Dyninst::MachRegister re
+        std::function<bool(int, std::string)> parse_attr_string)
+{
+    if (attr_string_size == 0) {
+        return false;
+    }
+
+    int curr = 0;
+    // The first character is the version of the attributes
+    // Currently only version 1, (aka 'A') is recognised
+    if (attr_string[curr] != 'A') {
+        create_printf("%s[%d]:  Unknown attributes version '%c'(%d) - expecting 'A'\n",
+                FILE__, __LINE__, attr_string[curr], attr_string[curr]);
+        return false;
+    }
+    curr++;
+
+    // Parse all attributes in the section
+    // The section can contain multiple "attribute sections" stacked together
+    // Each "attribute section" contains:
+    //     1. A 32-bit integer indicating the size of the current attribute section
+    //     2. A null terminated string, indicating the name of the current attribute
+    //     3. A series of "attributes". Each "attribute" contains
+    //         1) The tag number (1 => file attribute, 2 => section attribute, 3 => symbol attribute)
+    //         2) A 32-bit integer indicating the size of the attribute
+    //         3) A series of "attribute data". The attribute data is platform dependent
+
+    int section_end = attr_string_size;
+
+    // Parse all attribute sections
+    while (curr < section_end) {
+
+        // Get the attribute section size
+        int32_t attr_section_size = read_memory_as<int32_t>(&attr_string[curr]);
+        if (attr_section_size > section_end) {
+            create_printf("%s[%d]:  Bad attribute section length (%d > %d)\n",
+                    FILE__, __LINE__, attr_section_size, section_end);
+            return false;
+        }
+        int attr_section_end = curr + attr_section_size;
+        curr += sizeof(attr_section_size);
+        // The next null-terminated string indicates the name of the current attribute
+        const char *curr_attr_section_name = &attr_string[curr];
+        if (strcmp(curr_attr_section_name, attr_section_name) != 0) {
+            create_printf("%s[%d]:  Unexpected attribute section '%s'\n", FILE__, __LINE__, curr_attr_section_name);
+            return false;
+        }
+        curr += strlen(&attr_string[curr]) + 1;
+
+        // Parse all attributes in the attribute section
+        while (curr < attr_section_end) {
+            char tag = attr_string[curr];
+            curr++;
+
+            if (tag != 1) { // File attribute
+                create_printf("%s[%d]:  Unexpected tag %d\n",
+                        FILE__, __LINE__, tag);
+                return false;
+            }
+
+            int32_t attr_size = read_memory_as<int32_t>(&attr_string[curr]);
+            if (attr_size > attr_section_end) {
+                create_printf("%s[%d]:  Bad attribute length (%d > %d)\n",
+                        FILE__, __LINE__, attr_size, attr_section_end);
+                return false;
+            }
+            int attr_end = curr + attr_size - 1;
+            curr += sizeof(int);
```

Comment:
Sorry, I missed this one in my other review. Let's replace it, as well.

```suggestion
            curr += sizeof(attr_size);
```